server/src/main/java/org/opensearch/transport/BaseInboundMessage.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.transport;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * Base class for inbound data as a message.
+ * Different implementations are used for different protocols.
+ *
+ * @opensearch.internal
+ */
+@ExperimentalApi
+public interface BaseInboundMessage {
+
+    /**
+     * The protocol used to encode this message
+     */
+    static String NATIVE_PROTOCOL = "native";
```

Comment:
Belongs in a separate place/class/enum next to `determineProtocol()`.

Reply:
In the original PR, @reta had a comment for using string in place of enum to make it extensible. https://github.com/opensearch-project/OpenSearch/pull/11910#discussion_r1467990062. LMK what you think!

Reply:
@dblock for the context - enum restricts the protocol choices right away, with more generic approach (like strings), we could make protocols pluggable essentially.